### ISSUE TITLE
Save/get original number format or set isSupervisor.

### ISSUE THREAD

**Issue by AshMorrow** [PHPOffice/PhpSpreadsheet#1606]
This is:
```
- [ ] a bug report
- [x] a feature request
- [ ] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)
```
Cell have number format with leading zero
<a href="https://ibb.co/YLYX4Mt"><img src="https://i.ibb.co/x7PYK0M/2020-08-05-13-00.png" alt="2020-08-05-13-00" border="0"></a>

When i pars file in NumberFromat->setFormatCode get correct value, but this value not in 'builtInFormats' and i loose original format.
<a href="https://ibb.co/s3Xw2Rd"><img src="https://i.ibb.co/XJGsbLr/2020-08-05-13-06.png" alt="2020-08-05-13-06" border="0"></a>

How i can see,  I will get correct behavior if i set isSupervisor  but i dont see way to do that using `load`.

Also i can convert format using 'NumberFormat::toFormattedString', but i cant get original number format.

Sorry for my English.

**Comment by oleibman** [comment]
Doesn't getFormatCode do what you want?

**Comment by AshMorrow** [comment]
Yes, but no, setFormatCode will return wrong value. Usually it 4 and it is problem because i need something like "000000".

**Comment by oleibman** [comment]
I still don't understand your problem. setFormatCode returns $this, not the format code. On the other hand, getFormatCode does what you want. If I load a spreadsheet with Cell A1 formatted with your '000000' format, the following code works exactly as expected.
use PhpOffice\PhpSpreadsheet\Reader\Xlsx as ReaderXlsx;
use PhpOffice\PhpSpreadsheet\Writer\Xlsx as WriterXlsx;
$reader = new ReaderXlsx();
$file = 'xnumfmt2.xlsx';
$spreadsheet = $reader->load($file);
$sheet = $spreadsheet->getActiveSheet();
$numFmt = $sheet->getCell('A1')->getStyle()->getNumberFormat()->getFormatCode();
 //echo "numFmt is $numFmt\n"; // shows 000000 when not commented out
$sheet->getCell('A2')->setValue(31)
    ->getStyle()->getNumberFormat()->setFormatCode($numFmt);
$writer = new WriterXlsx($spreadsheet);
$writer->save('xnumfmt3.xlsx');

Cell A2 in the new spreadsheet is formatted exactly as you wish.

**Comment by AshMorrow** [comment]
<a href="https://ibb.co/g9L8j5v"><img src="https://i.ibb.co/DWHFVxG/2020-08-10-10-19.png" alt="2020-08-10-10-19" border="0"></a>

In my case getFormateCode return "General"

**Comment by oleibman** [comment]
I think I finally see your problem. You have created the spreadsheet with LibreOffice, not Excel, and, when PhpSpreadsheet reads it, it doesn't find the Number Format. I believe (could be wrong) that this is because LibreOffice isn't saving the formatting information. When I look at how the cell is saved, here's what I see (hope this system doesn't try to interpret the xml tags):
<table:table-column table:style-name="co1" table:default-cell-style-name="ce1"/>
<table:table-row table:style-name="ro1">
<table:table-cell office:value-type="float" office:value="123" calcext:value-type="float">
<text:p>000123</text:p>
</table:table-cell>
</table:table-row>
</table:table>
It looks like it has saved the text as '000123', with no formatting information associated with it, and so PhpSpreadsheet has nothing to go on as to how the cell was formatted. PhpSpreadsheet does handle some number formats from Libre Office when it can find them (percentage, currency, date, and time), but this is probably restricted to "builtin" formats, and only a very small number of those.

**Comment by oleibman** [comment]
I *was* wrong. The default-cell-style-name ce1 in content.xml points to a style N121 in styles.xml:
<number:number-style style:name="N121">
<number:number number:decimal-places="0" loext:min-decimal-places="0" number:min-integer-digits="6"/>
</number:number-style>
That is your format. And PhpSpreadsheet has some gaps, including this one, when processing styles from Libre Office. It is difficult because the references usually go through multiple layers of indirection, as this one does.

**Comment by oleibman** [comment]
Reopening. PR #4806 does not cover this situation, but lays the groundwork for doing so.
